ha/fault_tolerance/fault_tolerance_driver.py

```diff
@@ -14,42 +14,57 @@
 # For any questions about this software or licensing,
 # please email opensource@seagate.com or cortx-questions@seagate.com.
 
+"""
+   Set of periodically executable routines which accepts fault events from
+   message bus and takes further action on it
+"""
+
 
 import time
 
 from cortx.utils.conf_store import Conf
 from cortx.utils.log import Log
+from cortx.utils.message_bus import MessageConsumer
 
-from ha import const
 from ha.core.config.config_manager import ConfigManager
+from ha import const
 from ha.k8s_setup.const import _DELIM
+from ha.core.event_analyzer.filter.filter import K8SFilter
 
 
-class FaultToleranceDriver:
+class FaultTolerant:
     """
-    Driver class which will poll contineously with a specific
-    time gap and analyzes the alert condition and it gets
-    notified to event manager
+    Module responsible for consuming messages from message bus,
+    further analyzes that event and publishes it if required
     """
     def __init__(self, poll_time=10):
+        """Init method"""
+        self._poll_time = poll_time
+        self._k8s_filter = K8SFilter()
         ConfigManager.init('fault_tolerance_driver')
-        self._poll_time = Conf.get(const.HA_GLOBAL_INDEX, f"prometheus_config{_DELIM}poll_time")
-        if self._poll_time is None:
-            self._poll_time = poll_time
+        self._message_type = Conf.get(const.HA_GLOBAL_INDEX, f'MONITOR{_DELIM}message_type')
+        self._consumer = MessageConsumer(consumer_id='1', consumer_group='consumer-group', \
```

Comment:
we will register a callback function here;,  and anytime a message comes on messagebus the function will be called.

Reply:
Done.